Add recipe for AWS certificates

arnaudh opened the pull request:
This will allow us to use the AWS certificates directly in Julia, in order to verify the identity of RDS servers over SSL.

For example, using `LibPQ` to connect to an RDS PostgreSQL database, we'll be able to do:

```Julia
using LibPQ, AmazonCACerts_jll

sslrootcert = AmazonCACerts_jll.rds_ca_root_certificate
LibPQ.Connection("port=5432 dbname=test host=ec2-00-000-000-0.compute-1.amazonaws.com sslmode=verify sslrootcert=$sslrootcert")

```

Note: this is inspired by https://github.com/JuliaPackaging/Yggdrasil/tree/master/M/MozillaCACerts

giordano commented on A/AmazonCACerts/build_tarballs.jl:
```diff
@@ -0,0 +1,43 @@
+
+name = "AmazonCACerts"
+# Info here: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
+certificate_version = "2019"
+tarball_version = v"1.0.0"
```
We tend to use the same version number as the upstream "package", as we do in for `MozillaCACerts`: https://github.com/JuliaPackaging/Yggdrasil/blob/816c103b16c1b63fdb1ef17677be103c3efe3cb8/M/MozillaCACerts/build_tarballs.jl#L7-L8

arnaudh replied:
Oh, I must have done something wrong because I previously tried something similar and had an error, but not now.
Thanks, I'll update that.

giordano commented on A/AmazonCACerts/bundled/LICENSE:
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2020 Invenia Technical Computing Corporation
```
I'm a bit confused by this line: is this certificate produced by Invenia?

arnaudh replied:
Ah, I think I got confused here reading https://juliapackaging.github.io/BinaryBuilder.jl/dev/jll/ :
> LICENSE, a file stating the license of the JLL package. Note that this may differ from the license of the library it wraps, which is instead shipped inside the tarballs;

But now I realise this is for the `NAME_jll.jl` repo, which I understand will be generated after my PR gets merged.

I'll try to find the license for that AWS certificate and add it.

iamed2 commented on A/AmazonCACerts/bundled/LICENSE:
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2020 Invenia Technical Computing Corporation
```
This is actually the license for using the certs I believe and so this and the `install_license` line can be removed as well as the `DirectorySource`.

iamed2 commented on A/AmazonCACerts/build_tarballs.jl:
```diff
@@ -0,0 +1,43 @@
+sources = [
+    FileSource("https://s3.amazonaws.com/rds-downloads/rds-ca-$certificate_version-root.pem",
+    "d464378fbb8b981d2b28a1deafffd0113554e6adfb34535134f411bf3c689e73",
+    filename="rds-ca-root.pem"),
```
IMO we should keep the version in the file name so people can be aware of which version we're providing, in case we switch in the future. The variable name can remain the same.

giordano replied:
I don't have a strong opinion on this, but yeah, what you suggest is doable and totally fine.

arnaudh commented:
I've addressed the comments.

I removed the license since I couldn't find one for the AWS certificates. Now when I run `julia build_tarballs.jl` I see the following output:
```
$ julia build_tarballs.jl
[ Info: Building for any
┌ Error: Unable to find valid license file in "${prefix}/share/licenses/AmazonCACerts"
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/9oqch/src/Auditor.jl:438
```
However the exit code is 0 and the output `products/` is still correctly generated as far as I can tell.

giordano commented:
Yes, generally speaking we require a license, that isn't a fatal error but it uses `@error` to scare the user.  Is there any license that we can meaningfully use for the certificate?  For example `MozillaCACerts` has it.

arnaudh commented:
@giordano unfortunately I haven't found anything. Mozilla explicitly mentions a license [here](https://curl.haxx.se/docs/caextract.html), but Amazon does not ([here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) is the page listing the certificate; a little search on Google didn't uncover anything useful).

giordano commented:
Ok, well, let's move on for now
